.github/CODEOWNERS

```diff
@@ -1,5 +1,17 @@
-# These owners will be the default owners for everything in
-# the repo. Unless a later match takes precedence.
-*       @jmank88 @krebernisak
+# CODEOWNERS Best Practices
+# 1. Per Github docs: "Order is important; the last matching pattern takes the most precedence."
+# Please define less specific codeowner paths before more specific codeowner paths in order for the more specific rule to have priority
 
-pkg/monitoring/ @topliceanu @pedrovsg
+* @smartcontractkit/foundations
+
+/observability-lib/ @Atrax1
+
+/pkg/capabilities/ @cedric-cordenier @krehermann
+/pkg/codec @nolag
+/pkg/loop @jmank88 @krehermann
+/pkg/monitoring/ @topliceanu @pedrovsg
+/pkg/services @jmank88
+/pkg/sqlutil @jmank88
+/pkg/timeutil @jmank88
+/pkg/values/ @cedric-cordenier @krehermann
+/pkg/workflows/ @cedric-cordenier @krehermann
```

Comment:
Presence of a trailing `/` will not change what is matched, but we should be consistent to eliminate confusion. I suggest we require a trailing `/`

I think it's a small enough of a change and worth to do in this PR

Reply:
At a high level the code owner validator tool will also highlight the presence of individual ownership; we want to move to group based. 

@cedric-cordenier is there an appropriate github group for the keystone effort?

Reply:
@patrickhuie19 Yes, you could use this instead: https://github.com/orgs/smartcontractkit/teams/keystone

Reply:
Updated, but perhaps the team needs explicit access as a whole? 
![image](https://github.com/smartcontractkit/chainlink-common/assets/1194128/d0a60ca1-62f0-42b7-83f5-9c462744601c)

Reply:
Fixed